src/bindings/python/flux/asyncio/selector.py

```diff
@@ -0,0 +1,153 @@
+###############################################################
+# Copyright 2022 Lawrence Livermore National Security, LLC
+# (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+#
+# This file is part of the Flux resource manager framework.
+# For details, see https://github.com/flux-framework.
+#
+# SPDX-License-Identifier: LGPL-3.0
+###############################################################
+
+import selectors
+import signal
+
+from flux.constants import FLUX_POLLIN, FLUX_POLLOUT
+from flux.core.watchers import FDWatcher
+
+
+def stop_callback(handle, watcher, _, events, args=None):
+    """
+    This is called by the watcher, and stops the reactor add adds ready jobs.
+    """
+    _ = watcher
+    reactor = handle.get_reactor()
+    handle.reactor_stop(reactor)
+
+    # TimedWatcher won't know a file descriptor, just needs to stop the reactor
+    if not args:
+        return
+    selector = args["select"]
+    selector_event = get_selector_event(events)
+    selector.ready.add((args["key"], selector_event & args["key"].events))
+
+
+def get_selector_event(events):
+    """
+    Given an int, return the corresponding flux identifier.
+    """
+    event = 0
+    if events & FLUX_POLLIN:
+        event |= selectors.EVENT_READ
+    if events & FLUX_POLLOUT:
+        event |= selectors.EVENT_WRITE
+    return event
+
+
+def get_flux_event(events):
+    """
+    Given an int, return the corresponding selector identifier
+    """
+    event = 0
+    if events & selectors.EVENT_READ:
+        event |= FLUX_POLLIN
+    if events & selectors.EVENT_WRITE:
+        event |= FLUX_POLLOUT
+    return event
+
+
+class FluxSelector(selectors._BaseSelectorImpl):  # pylint: disable=protected-access
```

Comment:
yes! So I wanted the most bare selector that would be possible - without losing some useful functionality. If you look at the source code https://cs.github.com/python/cpython/blob/68c46ae68b6e0c36a12e37285fff9ce0782ed01e/Lib/selectors.py?q=_BaseSelectorImpl#L206 if we go one level, we lose nice courtesy functions in _BaseSelectorImpl that do the file descriptor lookup for us. We'd have to reimplement that, either copying it or something simpler (that maybe doesn't work as well). That also gives us `self.modify` which I haven't encountered hitting yet, but if we do hit it for some case that's what we'd want it to do (it's not implemented in the base class one level up). Finally, the next level down (SelectSelector and other variants) are tuned for specific kinds of events - we overwrite most of those classes so we could technically use it (because it's "public") but really we are using the _BaseSelectorImpl, which is the best match/level for what we need.

The distinction of it being "private" does indicate to external developers that it's not intended for this kind of use, however, so if you have strong feelings about that I can switch it to be the one level up public class, understanding that we aren't using most of it anyway.

Reply:
It sounds reasonable, but... why not the `DefaultSelector`? If the only trade-off is that we're overwriting some of its functionality. There's nothing wrong with some overwriting.

Reply:
DefaultScheduler is provided on the bottom - and you don't really know which one you are getting! Let's ask @trws about this one, I don't have a good sense of it it matters. I wanted to choose the most correct one for the job, which is exactly the `_BaseSelectorImpl` and the others are doing tweaks that we don't really need (and could lead to a bug, it's hard to say).

Reply:
I specifically requested to use _BaseSelectorImpl, at least while we're developing on this, because if we use DefaultSelector and miss something then (at least at first) things appeared to work when they were using the wrong loop, and yet _BaseSelectorImpl has some useful helpers in it.  We should probably switch to BaseSelector at some point.